### PR TITLE
fix context for beforeEach and afterEach hooks in Flow type definition file

### LINKED ISSUE #1340: Assigning `t.context` a value in `beforeEach` results in a Flow error
## Description

Assigning `t.context` a value in `beforeEach` results in a Flow error.

### Test Source

```js
// @flow

test.beforeEach(async (t) => {
  t.context = {};
});
```

### Error Message & Stack Trace

```
test/interceptors/cache/index.js:19
 19:   t.context = {};
       ^^^^^^^^^ assignment of property `context`. Property cannot be assigned on any member of intersection type
 19:   t.context = {};
       ^ intersection
  Member 1:
   83: type TestContext = AssertContext & {
                          ^^^^^^^^^^^^^ AssertContext. See: node_modules/ava/index.js.flow:83
  Error:
   19:   t.context = {};
           ^^^^^^^ property `context`. Property not found in
   83: type TestContext = AssertContext & {
                          ^^^^^^^^^^^^^ object type. See: node_modules/ava/index.js.flow:83
  Member 2:
   83: type TestContext = AssertContext & {
                                          ^ object type. See: node_modules/ava/index.js.flow:83
  Error:
   19:   t.context = {};
           ^^^^^^^ property `context`. Property not found in
   83: type TestContext = AssertContext & {
                                          ^ object type. See: node_modules/ava/index.js.flow:83
```

### Config

N/A

### Command-Line Arguments

N/A

## Environment

```
Node.js v7.7.1
darwin 16.5.0
0.18.2
4.1.2
```

### PR DESCRIPTION
**Pull request by zs-zs**
Fixes #1340

### CODE REVIEW

**Comment by novemberborn** [comment]
@gajus does this work for you?

**Comment by gajus** [comment]
This works. :+1:

**Comment by zs-zs** [comment]
I have no idea why it failed for node 4...

File: test/flow-types/regression-1114.js.flow [hunk 1]
```diff
@@ -25,11 +25,6 @@ test.cb(t => {
 	t.end();
 });
 
-test.beforeEach(t => {
-	// $ExpectError: Unknown property "context"
```
Comment (novemberborn) [inline test/flow-types/regression-1114.js.flow]: Perhaps keep the `beforeEach`, but remove this `$ExpectedError` line.

Comment (zs-zs) [inline test/flow-types/regression-1114.js.flow]: I thought about keeping the beforeEach, but then I thought it has nothing to with the issue #1114. But maybe it has, I didn't dig deeper.

Comment (zs-zs) [inline test/flow-types/regression-1114.js.flow]: BTW, how to trigger a rebuild? I didn't find that, never used Appveyor before.

Comment (novemberborn) [inline test/flow-types/regression-1114.js.flow]: It's used elsewhere in this file, so might be best to keep it.

I think you have to be in the AppVeyor team to do that.

Comment (sindresorhus) [inline test/flow-types/regression-1114.js.flow]: You can also push an empty commit or force push.

**Comment by novemberborn** [comment]
I think this is good now. I suppose the changes may break something because some types have been removed? But even if so it seems like a good bugfix that we should just ship in a `0.19.1` release.

**Comment by zs-zs** [comment]
Hm I don't know. I only removed "dead" types, so I hope it won't break anything.

**Comment by gajus** [comment]
Whats the shipping plan?

**Comment by novemberborn** [comment]
Thanks @zs-zs! I took out the dead type removal, we'll land that later: https://github.com/avajs/ava/pull/1352

**Pull request merged**